source/client-side-encryption/tests/unified/deleteKey.yml

```diff
@@ -0,0 +1,120 @@
+  - csfle: true
+
+createEntities:
+  - client:
+      id: &client0 client0
+  - clientEncryption:
+      id: &clientEncryption0 clientEncryption0
+      clientEncryptionOpts:
+        keyVaultClient: *client0
+        keyVaultNamespace: keyvault.datakeys
+        kmsProviders:
+          local: {}
+  - database:
+      id: &database0 database0
+      client: *client0
+      databaseName: &database0Name keyvault
+  - collection:
+      id: &collection0 collection0
+      database: *database0
+      collectionName: &collection0Name datakeys
+
+initialData:
+  - databaseName: *database0Name
+    collectionName: *collection0Name
+    documents:
+      - &aws_key_doc
+        _id: &aws_key_id { $binary: { base64: YXdzYXdzYXdzYXdzYXdzYQ==, subType: "04" } }
+        keyAltNames: ["aws_key"]
+        keyMaterial: { $binary: { base64: AQICAHhQNmWG2CzOm1dq3kWLM+iDUZhEqnhJwH9wZVpuZ94A8gFXJqbF0Fy872MD7xl56D/2AAAAwjCBvwYJKoZIhvcNAQcGoIGxMIGuAgEAMIGoBgkqhkiG9w0BBwEwHgYJYIZIAWUDBAEuMBEEDO7HPisPUlGzaio9vgIBEIB7/Qow46PMh/8JbEUbdXgTGhLfXPE+KIVW7T8s6YEMlGiRvMu7TV0QCIUJlSHPKZxzlJ2iwuz5yXeOag+EdY+eIQ0RKrsJ3b8UTisZYzGjfzZnxUKLzLoeXremtRCm3x47wCuHKd1dhh6FBbYt5TL2tDaj+vL2GBrKat2L, subType: "00" } }
+        creationDate: { $date: { $numberLong: "1641024000000" } }
+        updateDate: { $date: { $numberLong: "1641024000000" } }
+        status: 1
+        masterKey:
+          provider: aws
+          key: arn:aws:kms:us-east-1:579766882180:key/89fcc2c4-08b0-4bd9-9f25-e30687b580d0
+          region: us-east-1
+      - &local_key_doc
+        _id: &local_key_id { $binary: { base64: bG9jYWxrZXlsb2NhbGtleQ==, subType: "04" } }
+        keyAltNames: ["local_key"]
+        keyMaterial: { $binary: { base64: ABKBldDEoDW323yejOnIRk6YQmlD9d3eQthd16scKL75nz2LjNL9fgPDZWrFFOlqlhMCFaSrNJfGrFUjYk5JFDO7soG5Syb50k1niJoKg4ilsj0L4mpimFUtTpOr2nzZOeQtvAksEXc7gsFgq8gV7t/U3lsaXPY7I0t42DfSE8EGlPdxRjFdHnxh+OR8h7U9b8Qs5K5UuhgyeyxaBZ1Hgw==, subType: "00" } }
+        creationDate: { $date: { $numberLong: "1641024000000" } }
+        updateDate: { $date: { $numberLong: "1641024000000" } }
+        status: 1
+        masterKey:
+          provider: local
+
+tests:
+  - description: delete non-existent data key
+    operations:
+      - name: deleteKey
+        object: *clientEncryption0
+        arguments:
+          # *aws_key_id with first three letters replaced with 'A' (value: "3awsawsawsawsa").
+          id: { $binary: { base64: AAAzYXdzYXdzYXdzYXdzYQ==, subType: "04" } }
+        expectResult:
+          acknowledged: { $$unsetOrMatches: true }
+          deletedCount: 0
+    outcome:
+      - collectionName: *collection0Name
+        databaseName: *database0Name
+        documents:
+          - *aws_key_doc
+          - *local_key_doc
+
+  - description: delete existing AWS data key
+    operations:
+      - name: deleteKey
+        object: *clientEncryption0
+        arguments:
+          id: *aws_key_id
+        expectResult:
+          acknowledged: { $$unsetOrMatches: true }
```

Comment:
Is this necessary? If a write result was not acknowledged, all other fields would be empty. Therefore, it would not be possible to perform any assertion on `deletedCount`.

Reply:
`acknowledged` itself is optional (see: [Write Results](https://github.com/mongodb/specifications/blob/master/source/crud/crud.rst#write-results)). The reason for the [existing usages](https://github.com/mongodb/specifications/blob/7f37c9325b423223edf54d795afed444e6989b98/source/crud/tests/unified/deleteOne-hint-unacknowledged.yml#L66) is for asserting an unacknowledged write result. The CRUD spec permits methods to either return no result at all, or a document with an optional `acknowledged`. Also discussed in the CRUD spec: if an object has all optional fields, it may be optional itself. For that reason, we needed a convoluted way to handle an empty result or a possible `acknowledged: false` value. When `acknowledged` is `false`, "all other members of this result will be undefined."

In the case of an _acknowledged_ write result, there is really no reason to check for `acknowledged`. Other fields in the document will be accessible and can be asserted.

If you did want to assert `acknowledged`, then what you have here is correct since the field itself is optional. But I'm proposing we just remove the assertion for `acknowledged` altogether since an optional assertion for `true` is not done elsewhere (AFAIK) and is likely redundant.

Reply:
Thank you for the explanation. Given the field is optional, I am now confident it is written correctly as-is, as it is not permitted to omit the fielddue to matcher rules forbidding extra keys. For example, omitting the field entirely yields the following error in the C driver:
```
error: BSON match failed: match error at '': key acknowledged is not present
Expected:  { "acknowledged" : true, "deletedCount" : { "$numberInt" : "0" } }
Actual:  { "deletedCount" : { "$numberInt" : "0" } }
```

Reply:
Is your libmongoc branch up-to-date? Extra fields are permitted in root documents (see: [Unified Test Format spec](https://github.com/mongodb/specifications/blob/master/source/unified-test-format/unified-test-format.rst#allowing-extra-fields-in-root-level-documents)). I recently made a bug fix related to this in [CDRIVER-4374](https://jira.mongodb.org/browse/CDRIVER-4374).